src/dynamicCaller.ts

```diff
@@ -40,24 +48,41 @@ export class ServersNamespace {
 
   #performCall: PerformCallFn;
   #getTools: GetToolsFn;
+  #generatePrompt: GeneratePromptFn;
+  #getPrompts: GetPromptsFn;
 
   /**
    * Initialize the ServersNamespace with injected functions.
    *
    * @param performCall - Function to execute tool calls
    * @param getTools - Function to get tool schemas
+   * @param generatePrompt - Function to generate prompts
+   * @param getPrompts - Function to get prompt schemas
    */
-  constructor(performCall: PerformCallFn, getTools: GetToolsFn) {
+  constructor(
+    performCall: PerformCallFn,
+    getTools: GetToolsFn,
+    generatePrompt: GeneratePromptFn,
+    getPrompts: GetPromptsFn,
+  ) {
```

Comment:
if this keeps getting bigger we can change it to an object to make it easier to consume by its object keys instead of remembering which param goes in which position, I usually start doing that once I hit 3 or more args, but I'm also ok with keeping it like this for now

```suggestion
  constructor({
    performCall,
    getTools,
    generatePrompt,
    getPrompts,
  }) {
```